app/controllers/admin/stores_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Admin::StoresController < ApplicationController
+  def index
+    @stores = delegate_stores
+  end
+
+  def show
+    @store = Store.find(params[:id])
+  end
+
+  def delegate_stores
```

Comment:
can this be a private method? also you can do something like "return Store.where(params[:status] ? status: params[:status])" (Store.where is the same as Store.all). I personally am not the biggest fan of switching a function by the value, but thats just me